py/selenium/webdriver/common/proxy.py

```diff
@@ -284,13 +284,8 @@ def _verify_proxy_type_compatibility(self, compatible_proxy):
                 f"Specified proxy type ({compatible_proxy}) not compatible with current setting ({self.proxyType})"
             )
 
-    def add_to_capabilities(self, capabilities):
```

Comment:
Minor comment, wouldn't this break existing implementations?

Reply:
Yeah, this is the one contentious piece, which is why I explained my reasoning in the Motivation section above. I'm pretty sure it only breaks if people are using deprecated classes. This is the hard bit of trying to juggle changes in too many places; I was hoping to get #12030 in the same release and it wouldn't matter.

Reply:
I think the code is correct. I believe we should merge this, otherwise we will stall for longer on this.